### PR TITLE
sn32: shared matrix: switch to continuous pwm

### PR DESCRIPTION
**Pull request by dexter93**
## Description

Switch from oneshot PWM to continuous. This will enable full LED output and minimize any flickering.
Key matrix scan rate is also heavily improved from ~60ms avg down to ~25ms avg with lows at ~8ms
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: drivers/sn32/rgb_matrix_sn32f248b.c [hunk 1]
```diff
@@ -33,11 +33,9 @@
     (B)     (E)
     GPIO    GND
 */
-static uint16_t row_ofsts[LED_MATRIX_ROWS];
 static uint8_t mr_offset[24] = {0};
 static uint8_t current_row = 0;
 LED_TYPE led_state[LED_MATRIX_ROWS * LED_MATRIX_COLS];
```
Comment (IsaacDynamo) [inline drivers/sn32/rgb_matrix_sn32f248b.c]: `LED_MATRIX_ROWS * LED_MATRIX_COLS` should be changed to `DRIVER_LED_TOTAL`

File: drivers/sn32/rgb_matrix_sn32f248b.c [hunk 2]
```diff
@@ -355,15 +324,15 @@ void SN32F24XX_init(void) {
...
-    led_state[corrected_index].g = g;
-    led_state[corrected_index].b = b;
+    led_state[index].r = r;
+    led_state[index].g = g;
+    led_state[index].b = b;
 }
 
 void SN32F24XX_set_color_all(uint8_t r, uint8_t g, uint8_t b) {
-    for (int i=0; i<DRIVER_LED_TOTAL; i++)
+    for (int i = 0; i < sizeof(led_state) / sizeof(led_state[0]); i++) {
```
Comment (IsaacDynamo) [inline drivers/sn32/rgb_matrix_sn32f248b.c]: `sizeof(led_state) / sizeof(led_state[0])` can be changed back to `DRIVER_LED_TOTAL` when `led_state` is made `DRIVER_LED_TOTAL` in size.

File: drivers/sn32/rgb_matrix_sn32f248b.c [hunk 3]
```diff
@@ -355,15 +324,15 @@ void SN32F24XX_init(void) {
 static void flush(void) {}
 
 void SN32F24XX_set_color(int index, uint8_t r, uint8_t g, uint8_t b) {
```
Comment (IsaacDynamo) [inline drivers/sn32/rgb_matrix_sn32f248b.c:358]: Functions SN32F24XX_init(), SN32F24XX_set_color(), SN32F24XX_set_color_all() can be made static.
And should be removed from rgb_matrix_sn32f24xx.h
The SN32F24XX_ prefix is also no longer needed, and can be removed.

File: drivers/sn32/rgb_matrix_sn32f248b.c [hunk 4]
```diff
@@ -267,86 +265,57 @@ void shared_matrix_rgb_disable(void) {
         pwmDisableChannel(&PWMD1,i);
         chSysUnlockFromISR();
```
Comment (IsaacDynamo) [inline drivers/sn32/rgb_matrix_sn32f248b.c]: Why are the chSysLockFromISR() and chSysUnlockFromISR() calls needed here?

**Comment by dexter93** [comment]
Col2Row is good to go.

**Comment by dexter93** [comment]
Current shared matrix logic can be visualized here. https://tinyurl.com/y57nbk8s

**Comment by dexter93** [comment]
PR updated. Col2Row and Row2Col work properly now

**Comment by dexter93** [comment]
Reports say this takes care of https://github.com/SonixQMK/qmk_firmware/issues/157 too

**Pull request merged**